src/main/resources/avro/readmethods.avdl

```diff
@@ -72,7 +72,20 @@ record SearchReadsResponse {
 }
 
 /**
-Gets a list of `ReadAlignment` matching the search criteria.
+Gets a list of `ReadAlignment`s for one or more `ReadGroup`s.
+
+`searchReads` operates over a genomic coordinate space of reference sequence
+and position defined by the `Reference`s to which the requested `ReadGroup`s are
+aligned.
+
+If a target positional range is specified, search returns all reads whose
+alignment to the reference genome *overlap* the range. A query which specifies
+only read group IDs yields all reads in those read groups, including unmapped
+reads.
+
+All reads returned (including reads on subsequent pages) are ordered by genomic
+coordinate (by reference sequence, then position). Reads with equivalent genomic
+coordinates are returned in an unspecified but deterministic order.
```

Comment:
Please include the rationale for ordering as paging in the
doc.  Prevents asking the same questions about why.


CH Albach <notifications@github.com> writes:
> In src/main/resources/avro/readmethods.avdl:
> 
> > @@ -72,7 +72,20 @@ record SearchReadsResponse {
> >  }
> >
> >  /**
> > -Gets a list of `ReadAlignment` matching the search criteria.
> > +Gets a list of `ReadAlignment`s for one or more `ReadGroup`s.
> > +
> > +`searchReads` operates over a genomic coordinate space of reference sequence
> > +and position defined by the `Reference`s to which the requested `ReadGroup`s are
> > +aligned.
> > +
> > +If a target positional range is specified, search returns all reads whose
> > +alignment to the reference genome *overlap* the range. A query which specifies
> > +only read group IDs yields all reads in those read groups, including unmapped
> > +reads.
> > +
> > +All reads returned (including reads on subsequent pages) are ordered by genomic
> > +coordinate (by reference sequence, then position). Reads with equivalent genomic
> > +coordinates are returned in an unspecified but deterministic order.
> 
> What I was trying to express here was: if I send the same request twice to the
> same repository over data which hasn't changed, I expect to get the same
> ordering. I don't want clients to depend on one server's specific ordering, but
> rather give them the option of depending the property that there is a
> self-consistent ordering for one repository. I also wasn't really happy with
> this wording though - suggestions?
> 
> For example, say I'm creating a UI back by SearchReads which displays paginated
> results with pageSize=10, and the first matches are 100 different reads which
> are all mapped to the same position. If I refresh the page, I would like the
> first page to contain the same content every time - rather than a random
> permutation of 10 reads from those 100.
> 
> —
> Reply to this email directly or view it on GitHub.*
>